### PR TITLE
[export] Fix graph output mismatch issue with constant outputs.

### PR DESCRIPTION
**Pull request by zhxchen17**
Fixes #ISSUE_NUMBER

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @avikchaudhuri @gmagogsfm @tugsbayasgalan @angelayi @suo @ydwu4

### CODE REVIEW

File: torch/_dynamo/codegen.py [hunk 1]
```diff
@@ -98,6 +102,8 @@ def __call__(self, value, allow_cache=True):
                 self(value.mutable_local.source)
                 return
 
+        allow_cache &= not self.skip_cache
```
Comment (tugsbayasgalan) [inline torch/_dynamo/codegen.py:105]: maybe add a comment here about this being export specific case?

File: torch/_export/__init__.py [hunk 2]
```diff
@@ -351,6 +351,7 @@ def _get_params_buffers(mod: torch.nn.Module) -> Dict[str, torch.Tensor]:
     return params_buffers
 
 
+# TODO This wrapper should be removed.
```
Comment (tugsbayasgalan) [inline torch/_export/__init__.py:354]: Hmm??

**Review by tugsbayasgalan — APPROVED** [review]

File: test/export/test_export.py [hunk 3]
```diff
@@ -1542,6 +1542,32 @@ def foo(a, b):
...
+
+        mod = ModuleConstant()
+        ep = torch.export.export(mod, ())
+        self.assertEqual(ep(), mod())
+
+        args = (torch.randn(3, 2), torch.randn(3, 2))
+        mod = ModuleNestedConstant()
+        ep = torch.export.export(mod, args)
+        self.assertEqual(ep(*args), mod(*args))
+
```
Comment (cccclai) [inline test/export/test_export.py:1570]: Can we add two more test cases as they were the one errors out before?

```
    def test_empty_input(self):
        def g():
            return torch.zeros(1, 2)

        pre_autograd_graph = torch._export.capture_pre_autograd_graph(g, ())
        export_graph = torch.export.export(g, ())

    def test_scalar_input(self):
        def g(x):
            return torch.zeros(1, x)

        pre_autograd_graph = torch._export.capture_pre_autograd_graph(g, (2, ))
        # export_graph = torch.export.export(g, (2, ))
```

**Pull request closed without merging**